File: book-service/src/main/java/com/bookservice/exception/GeneralExceptionHandler.java
package com.bookservice.exception;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

@RestControllerAdvice
public class GeneralExceptionHandler extends ResponseEntityExceptionHandler {

    @ExceptionHandler(BookNotFoundException.class)
    public ResponseEntity<Object> handleBookNotFoundException(BookNotFoundException ex) {
        return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
    }

    @ExceptionHandler(NotFoundException.class)
    public ResponseEntity<Object> handleAuthorNotFoundException(NotFoundException ex) {
        return new ResponseEntity<>(ex.getExceptionMessage(), HttpStatus.NOT_FOUND);
    }
}

File: author-service/src/main/kotlin/authorservice/exception/GeneralExceptionHandler.kt
package authorservice.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler

import java.util.Objects

@RestControllerAdvice
class GeneralExceptionHandler : ResponseEntityExceptionHandler() {

    @ExceptionHandler(AuthorNotFoundException::class)
    fun handleReviewNotFoundException(ex: AuthorNotFoundException): ResponseEntity<String> {
        return ResponseEntity.status(404).body(Objects.requireNonNull(ex.message))
    }
}
